File: 프로그래머스/lv1/12926. 시저 암호/시저 암호.py
def solution(s, n):
    answer = list()

    for letter in s:
        if letter.isalpha():
            temp = ord(letter)
            if letter.isupper():
                if temp + n <= 90:
                    answer.append(chr(temp + n))
                else:
                    answer.append(chr(temp + n - 26))
            else:
                if temp + n <= 122:
                    answer.append(chr(temp + n))
                else:
                    answer.append(chr(temp + n - 26))
        else:
            answer.append(letter)

    return (''.join(answer))

File: 프로그래머스/1/12926. 시저 암호/시저 암호.cpp
#include <string>
#include <cctype>

using namespace std;

string solution(string s, int n) {
    for (int i = 0; i < s.length(); i++) {
        if (!isalpha(s[i])) continue;
        
        if (isupper(s[i]) && (s[i] - 'A' + n >= 26) ||
            islower(s[i]) && (s[i] - 'a' + n >= 26)) {
            s[i] += n - 26;
        } else {
            s[i] += n;
        }
    }
    
    return s;
}